gather_gitops.sh

```diff
@@ -0,0 +1,177 @@
+    done
+
+    echo -e "getting services in "${namespace}"..."
+    mkdir -p ""${gitops_folder}"/namespace_"${namespace}"_resources/services"
+    for service in $(oc get services -n "${namespace}" -o jsonpath='{ .items[*].metadata.name }') ; do
+      oc -n "${namespace}" get "${service}" -o yaml > ""${gitops_folder}"/namespace_"${namespace}"_resources/services/"${service}".yaml" 2>&1
+    done
+
+    echo -e "getting replicaSets in "${namespace}"..."
+    mkdir -p ""${gitops_folder}"/namespace_"${namespace}"_resources/replicasets"
+    for replicaset in $(oc get replicasets -n "${namespace}" -o jsonpath='{ .items[*].metadata.name }') ; do
+      oc -n "${namespace}" get "${replicaset}" -o yaml > ""${gitops_folder}"/"namespace_"${namespace}"_resources"/replicasets/"${replicaset}".yaml" 2>&1
+    done
+
+    echo -e "getting statefulsets in "${namespace}"..."
+    mkdir -p ""${gitops_folder}"/namespace_"${namespace}"_resources/statefulsets"
+    for statefulset in $(oc get statefulsets -n "${namespace}" -o jsonpath='{ .items[*].metadata.name }') ; do
+      oc -n "${namespace}" get "${statefulset}" -o yaml > ""${gitops_folder}"/"namespace_"${namespace}"_resources"/statefulsets/"${statefulset}".yaml" 2>&1
+    done
+
+    echo -e "getting routes in "${namespace}"..."
+    mkdir -p ""${gitops_folder}"/namespace_"${namespace}"_resources/routes"
+    for route in $(oc get routes -n "${namespace}" -o jsonpath='{ .items[*].metadata.name }') ; do
+      oc -n "${namespace}" get "${route}" -o yaml > ""${gitops_folder}"/namespace_"${namespace}"_resources/routes/"${route}".yaml" 2>&1
+    done
+
+    echo -e "getting ArgoCD in "${namespace}"..."
+    oc -n "${namespace}" get argocd -o yaml > ""${gitops_folder}"/namespace_"${namespace}"_resources/argocd.yaml" 2>&1
+
+    echo -e "getting Applications in "${namespace}"..."
+    mkdir -p ""${gitops_folder}"/namespace_"${namespace}"_resources/applications"
+    for application in $(oc get applications.argoproj.io -n "${namespace}" -o jsonpath='{ .items[*].metadata.name }') ; do
+      oc -n "${namespace}" get "${application}" -o yaml > ""${gitops_folder}"/namespace_"${namespace}"_resources/applications/"${application}".yaml" 2>&1
+    done
+
+    local sourceNamespaces=$(oc get argocd -n "${namespace}" -o jsonpath='{.items[*].spec.sourceNamespaces[*]}' )
+    if [[ "${sourceNamespaces}" != "" ]] ; then
+      for sourceNamespace in ${sourceNamespaces} ; do 
+        local sourceNamespaceApps=$(oc get applications.argoproj.io -n "${sourceNamespace}" -o jsonpath='{ .items[*].metadata.name }' )
+        for sourceNamespaceApp in ${sourceNamespaceApps}; do 
+          oc -n "${sourceNamespace}" get "${sourceNamespaceApplication}" -o yaml > ""${gitops_folder}"/namespace_"${namespace}"_resources/applications/"${sourceNamespaceApp}"_sourceNamespace_"${sourceNamespace}".yaml" 2>&1
```

Comment:
I guess you meant `oc -n "${sourceNamespace}" get "${sourceNamespaceApp}"` here.